### DISCUSSION TITLE
Adding Keda to an Istio service mesh

### DISCUSSION THREAD

**Discussion by haim-brum** [kedacore/keda#3840] [Q&A / Need Help]
As described in the docs, Keda can't be a part of an Istio service mesh, and if there is a service mesh on the cluster we should exclude Keda namespace by setting the label istio-injection=disabled.
My issue is that I have defined `PeerAuthentication` policy to enforce TLS communication over the cluster.
I created a `ScaledObject` with Prometheus trigger in order to auto-scale by some Prometheus metrics.
If I will keep the `PeerAuthentication` while excluding Keda from the service mesh, Istio will block any non TLS connections to Prometheus, which will break the scaler.
Is there a way to take benefits of Istio service mesh without compromising the TLS enforcement?

ScaledObject
```
apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: express
  namespace: test-app
spec:
  scaleTargetRef:
    name: express       
  pollingInterval:  10 
  cooldownPeriod:   300
  minReplicaCount:  1  
  maxReplicaCount:  10 
  triggers:
  - type: prometheus
    metadata:
      serverAddress: http://kube-prometheus-stack-prometheus.monitoring:9090
      metricName: http_requests_total
      query: http_requests_total{container="express"}
      threshold: '10'
```
Keda namespace
```
Name:         keda
Labels:       istio-injection=disabled
              kubernetes.io/metadata.name=keda
Annotations:  <none>
Status:       Active
```
Istio PeerAuthorization policy
```
apiVersion: security.istio.io/v1beta1
kind: PeerAuthentication
metadata:
  name: "default"
  namespace: "istio-system"
spec:
  mtls:
    mode: STRICT
```

keda-operator-metrics-apiserver logs
```
E1110 10:47:57.548771       1 prometheus_scaler.go:294] keda_metrics_adapter/prometheus_scaler "msg"="error executing prometheus query" "error"="Get \"http://kube-prometheus-stack-prometheus.monitoring:9090/api/v1/query?query=http_requests_total%7Bcontainer%3D%22express%22%7D&time=2022-11-10T10:47:57Z\": read tcp 172.17.0.16:45722->10.96.169.53:9090: read: connection reset by peer" "name"="express" "namespace"="test-app" "type"="ScaledObject"
E1110 10:47:57.552349       1 provider.go:124] keda_metrics_adapter/provider "msg"="error getting metric for scaler" "error"="Get \"http://kube-prometheus-stack-prometheus.monitoring:9090/api/v1/query?query=http_requests_total%7Bcontainer%3D%22express%22%7D&time=2022-11-10T10:47:57Z\": read tcp 172.17.0.16:45722->10.96.169.53:9090: read: connection reset by peer" "scaledObject.Name"="express" "scaledObject.Namespace"="test-app" "scaler"={}
E1110 10:47:57.552383       1 status.go:71] apiserver received an error that is not an metav1.Status: &errors.errorString{s:"no matching metrics found for s0-prometheus-http_requests_total"}: no matching metrics found for s0-prometheus-http_requests_total
```

Thank you.

**Comment by JorTurFer** [comment]
The problem with service mesh is that HPA controller has to reach the KEDA metrics server, if you enforce all the KEDA traffic through the service mesh, the HPA Controller won't reach KEDA. IDK if you can allow that behaviour IDK how because I don't have any expertise with service meshes.
Other option could be just using https in KEDA, but maybe you face with certificate validation errors because I guess that the tls certificate is self-signed, right? There are some scalers which implement support to ignore the certificate validation, so if you are willing to contribute, you could add the support to skip tls certificate validation and refer the https endpoint in Prometheus scaler.
WDYT?
